sherpa/models/tests/test_parameter.py

```diff
@@ -220,11 +223,12 @@ def test_binop_string():
     '''
     p, p2 = setUp_composite()
     comp = p + p2
-    assert repr(comp) == "<BinaryOpParameter '(model.p + model.p2)'>"
+    assert repr(comp) == "<BinaryOpParameter 'model.p + model.p2'>"
     comp = 1 + p * 2.7**p2
-    assert repr(comp) == "<BinaryOpParameter '(1 + (model.p * (2.7 ** model.p2)))'>"
+    assert repr(comp) == "<BinaryOpParameter '1 + model.p * 2.7 ** model.p2'>"
     comp = np.logaddexp2(p, p2)
-    assert repr(comp) == "<BinaryOpParameter 'numpy.logaddexp2(model.p, model.p2)'>"
+    assert repr(comp) == "<BinaryOpParameter 'numpy.logaddexp2(model.p, (model.p2))'>"
```

Comment:
This is removing some many brackets, a few more here and there are still worth it. I'm just wandering: Are the new () around `model.p2` design intent or an artifact of the implementation? I'm OK with either (better to keep the implementation simple at the cost of a few extra `()`, I'm just wondering.

Reply:
I believe it is an artifact of the decision I make to not try to be clever with unary-op models, **but** I can not guarantee that. I'll have to look.

Reply:
So, it's an interesting decision in sherpa.utils.get_precedence_rhs which is deciding between ignoring brackets when you have, for a given operator `<op>`

   ... <op> term

For addition and multiplication we don't add the brackets around `term` but for everything else we do. However, this is only relevant for binary expressions written as `lterm <op> rterm`, and `np.logaddexp2` is an example of "do not need ot bother". So I'll have a think about this when I move the code to `sherpa.utils.op_or_some_other_name`

Reply:
Now I've created `sherpa.models.op` I need to look into this (but wanted to get the updated code tested on CI before trying to do this).

Reply:
I've special-cased the `BinaryOpParameter` so that it only bothers with the bracket removal for infix operators, since prefix operators like `np.foobar(lhs, rhs)` do not need special-casing the `lhs` and `rhs` terms.